pkg/spec/deployment.go

```diff
@@ -54,7 +54,7 @@ func (deployment *DeploymentSpecMod) Fix() error {
 		return errors.Wrap(err, "unable to fix ControllerFields")
 	}
 
-	addKeyValueToMap(appLabelKey, deployment.ControllerFields.Name, deployment.ObjectMeta.Labels)
+	deployment.ControllerFields.ObjectMeta.Labels = addKeyValueToMap(appLabelKey, deployment.ControllerFields.Name, deployment.ControllerFields.ObjectMeta.Labels)
```

Comment:
👍 this is good. Returning new map is much better. This is much more readable, It is possible to see what is happening on this line even from the first look.